### PR TITLE
clean external auth; see #1548; should fix #1611

### LINKED ISSUE #1611: [GLPI 9.1.1] logout will re-login automatically when using SSO with LDAP
Hello,

In the past there was a parameter on the logout url that was noAuto=1, where is gone this parameter?

image

Now with today sources (from 9.1/bugfixes), this option seems to have disappear, and of course there is no way to logout, as the logout will re-login automatically and immediately...

info: authtype is 3: LDAP directory

Thank you to re-enable this option on the logout url.

### PR DESCRIPTION
**Pull request by orthagh**
Ok, another change into auth system needed with the @tomolimo report in #1611.

A resume, in https://github.com/glpi-project/glpi/blob/master/inc/auth.class.php#L581
We are in a condition matching only alternate auth (l581).
So in the lower block if/else, we can never execute "else" part.

The noAuto=1 on logout requires $_SESSION['glpiextauth'] == true
see : https://github.com/glpi-project/glpi/blob/master/inc/html.class.php#L1430
and https://github.com/glpi-project/glpi/blob/b4f97248b3f7da1c787e94cceda6a25edde415bf/inc/session.class.php#L116

In case of autologin, we need to set $this->extauth = 1 to avoid relogin on logout.

IMO, the case of blanking password is needed when when change authtype of a user, so i completely unlink it from authtype or login.

Last changes:
- Api is not an external auth, i think
- Could we always set noAuto=1 in case of logout ?

### CODE REVIEW

File: inc/auth.class.php [hunk 1]
```diff
@@ -1033,7 +1024,7 @@ static function useAuthExt() {
     * @return boolean
    **/
    static function isAlternateAuth($authtype) {
-      return in_array($authtype, array(self::X509, self::CAS, self::EXTERNAL, self::API));
+      return in_array($authtype, array(self::X509, self::CAS, self::EXTERNAL));
```
Comment (remicollet) [inline inc/auth.class.php]: > Api is not an external auth, i think

This function is about "Alternate" not "External"... always the same confusion...

**Comment by remicollet** [comment]
> Could we always set noAuto=1 in case of logout ?

Seems a simpler way.

**Comment by tomolimo** [comment]
> Could we always set noAuto=1 in case of logout ?
Seems a simpler way.

This is also ok for me: it is the behavior we had on the 0.83.8 :)

File: inc/auth.class.php (hunk 1 above)
Comment (orthagh) [inline inc/auth.class.php]: right, i change this part.

**Comment by orthagh** [comment]
it was my question in fact (maybe not very clear), why this changed was added ?

**Comment by trasher** [comment]
@tomolimo could you please tests the whole fix in your environment?

Just to be sure ;)

**Comment by trasher** [comment]
Great, thank you!

**Comment by tomolimo** [comment]
you're welcome :)

**Pull request merged**